contracts/access/access-control/traits.rs

```diff
@@ -18,40 +19,12 @@ use ink_storage::traits::StorageLayout;
 #[derive(Default, Debug, SpreadLayout)]
 #[cfg_attr(feature = "std", derive(StorageLayout))]
 pub struct AccessControlData {
-    pub roles: BTreeMap<RoleType, RoleData>,
+    pub admin_roles: StorageHashMap<RoleType, RoleType>,
+    pub members: StorageHashMap<(RoleType, AccountId), ()>,
```

Comment:
Because it is hash map and it is faster than binary search with`log(n)`.
Another point is that we need support ordering in vector, and it is hard to do in the case of storage(it will do a lot of load/save operations).